Assets/GothicVR/Scripts/Npc/Actions/AnimationActions/GoToWp.cs

```diff
@@ -21,13 +22,20 @@ public override void Start()
              * 1. AI_StartState() can get called multiple times until it won't share the WP. (e.g. ZS_SLEEP -> ZS_StandAround())
              * 2. Happens (e.g.) during spawning. As we spawn NPCs onto their current WayPoints, they don't need to walk there from entrance of OC.
              */
-            if (destination == "" || Props.CurrentWayPoint.Name == destination)
+            if (Props.CurrentWayPoint != null && (destination == "" || Props.CurrentWayPoint.Name == destination))
             {
                 IsFinishedFlag = true;
                 return;
             }
 
-            route = new Stack<DijkstraWaypoint>(WayNetHelper.FindFastestPath(Props.CurrentWayPoint.Name, destination));
+            WayPoint waypoint = Props.CurrentWayPoint != null ? Props.CurrentWayPoint : WayNetHelper.FindNearestWayPoint(Props.transform.position);
```

Comment:
Again: CWayPoint == null. Can we fix this earlier? Seems like an unstable state.

Reply:
it's better to check against wrong script code by checking if currentwaypoint is null and also if destination is correct